android/app/build.gradle

```diff
@@ -151,6 +151,7 @@ dependencies {
     implementation 'androidx.browser:browser:1.7.0'
     implementation 'androidx.activity:activity-ktx:1.8.2'
     implementation "androidx.work:work-runtime:2.9.0"
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-guava:1.6.0"
```

Comment:
Do the coroutines imports in 164-166 not offer the functionality you need? If not then can you move this impl to that coroutines section